aemcmc/negative_binomial_horseshoe_gibbs.py

```diff
@@ -32,8 +34,47 @@ def update_beta(rng, omega, lambda2tau2_inv, X, z):
     )
 
 
-# NOTE: invGamma(1, a) === 1 / Exp(a), where a is the scale/mean of the exponential
 def horseshoe_step(srng, beta, sigma2, lambda2_inv, tau2_inv):
+    r"""Gibbs kernel to sample from the posterior distribution of a horsehoe prior.
+
+    This kernel generates samples from the posterior distribution of the local
+    and global shrinkage parameters of a horsehoe prior, respectively :math:`\lambda`
+    and :math:`\tau` in the following model:
+
+    .. math::
+
+        \begin{align*}
+            \beta_j &\sim \operatorname{Normal}(0, \lambda_j^2\;\tau^2\;\sigma^2)\\
+            \sigma^2 &\sim \sigma^{-2} \mathrm{d} \sigma\\
+            \lambda_j^2 &\sim \operatorname{HalfCauchy}(0, 1)\\
```

Comment:
```suggestion
            \lambda_j &\sim \operatorname{HalfCauchy}(0, 1)\\
```
The parameter isn't squared in the model specification, only when its reparameterized into the scale-mixture representation. In fact, I would use that representation here for clarity since the half-cauchy representaion isn't used directly.

Reply:
Thank you!

> In fact, I would use that representation here for clarity since the half-cauchy representation isn't used directly.

I see your point, but the point of the docstring is not to describe the code but the model to which this sampler typically applies. This is the common representation of a horseshoe prior.

This makes me think that we will need a way to represent these equivalences between representations.